### ISSUE TITLE
Client: player:set_look_dir() lags a client-step behind player:set_pos() 

### ISSUE THREAD

**Issue by Warr1024** [minetest/minetest#12337]
##### Minetest version

```
Minetest 5.5.0 (Linux)
Using Irrlicht 1.9.0mt4
Using Lua 5.1.5
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=0
USE_SOUND=1
STATIC_SHAREDIR="."
```

##### OS / Hardware

Operating system: Linux 5.10.0-14-amd64 https://github.com/minetest/minetest/pull/1 SMP Debian 5.10.113-1 (2022-04-29) x86_64 GNU/Linux
CPU: AMD EPYC 7282 16-Core Processor

##### Summary

In the **szutil_cinecam** mod (used for the NodeCore livestream), I want to cut from one camera position to another (including the camera direction).  However, no matter what I do, it seems to always move the camera to the new position but *old* angle first, and then change the camera angle on a different step.  This creates often visible jank in the stream.

It seems like this might be due to the pos/angle packets being processed on different client steps, especially since this seems to happen more in areas with more complex geometry.

##### Steps to reproduce

Code: https://gitlab.com/sztest/szutilpack/-/blob/64f28532f1027d559363a80e6c87cbb59fb2f5bd/szutil_cinecam/init.lua#L90-94 (note that the behavior is similar even if the angle-setting and position-setting code is swapped in order).

To test locally:
* Install https://content.minetest.net/packages/Warr1024/szutilpack/
* Enable szutil_cinecam in a world
* /grantme fly,noclip,szutil_cinecam
* Enable fly mode to prevent the "camera" (player) from falling

You can also watch the livestream at https://www.twitch.tv/nodecoremt and see the camera transitions (at least a couple times per minute).  When players are on, some of them may be in areas that render smoothly, so you might notice little or no camera jank in the jumpcuts (sometimes only a frame or so), but especially when there are no players on, and the screen dims and the camera just moves around random positions near spawn, some of the hidden geometry in that area will slow down the transition enough that you can clearly see the double-cut.

**Comment by Zughy** [comment]
Duplicate of #12292 ?

**Comment by Warr1024** [comment]
> Duplicate of #12292 ?

No, looks like not a duplicate.  That one seems to be about client pos lagging behind server pos, while this is about lookdir lagging behind pos.  They could be different facets of some single underlying problem, but until the cause is identified, these are 2 separate symptoms and might still have 2 different causes.

**Comment by Zughy** [comment]
I haven't tested locally, but the code seems fine and I've noticed the jumps you were referring to in the livestream
